src/librustc_resolve/lib.rs

```diff
@@ -2077,40 +2086,42 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
             this.visit_generics(generics);
```

Comment:
(well, my attempt to quickly hack this in was met with failure: `internal compiler error: self type has not been fully resolved`)